src/renderers/native/ReactNativeFiber.js

```diff
@@ -0,0 +1,367 @@
+    // Either way we need to pass a copy of the Array to prevent it from being frozen.
+    const nativeTags = parentInstance._children.map(
+      (child) => typeof child === 'number'
+        ? child // Leaf node (eg text)
+        : child._nativeTag
+    );
+
+    UIManager.setChildren(
+      parentInstance._nativeTag, // containerTag
+      nativeTags // reactTags
+    );
+  },
+
+  getChildHostContext(
+    parentHostContext : string | null,
+    type : string
+  ) {
+    return parentHostContext;
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    const beforeChildIndex = parentInstance._children.indexOf(beforeChild);
+    const index = parentInstance._children.indexOf(child);
+
+    // Move existing child or add new child?
+    if (index >= 0) {
+      parentInstance._children.splice(index,1);
+      parentInstance._children.splice(beforeChildIndex,0,child);
+
+      UIManager.manageChildren(
+        parentInstance._nativeTag, // containerID
+        [index], // moveFromIndices
+        [beforeChildIndex], // moveToIndices
+        [], // addChildReactTags
+        [], // addAtIndices
+        [], // removeAtIndices
+      );
+    } else {
+      parentInstance._children.splice(beforeChildIndex,0,child);
+
+      UIManager.manageChildren(
+        parentInstance._nativeTag, // containerID
+        [], // moveFromIndices
+        [], // moveToIndices
+        [child._nativeTag], // addChildReactTags
+        [beforeChildIndex], // addAtIndices
+        [], // removeAtIndices
+      );
+    }
+  },
+
+  prepareForCommit() : void {
+    // Noop?
+  },
+
+  prepareUpdate(
+    instance : Instance,
+    oldProps : Props,
+    newProps : Props
+  ) : boolean {
+    return true;
+  },
+
+  removeChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    recursivelyUncacheFiberNode(child);
+
+    if (typeof parentInstance === 'number') {
+      UIManager.manageChildren(
+        parentInstance, // containerID
+        [], // moveFromIndices
+        [], // moveToIndices
+        [], // addChildReactTags
+        [], // addAtIndices
+        [0], // removeAtIndices
+      );
+    } else {
+      const index = parentInstance._children.indexOf(child);
+
+      parentInstance._children.splice(index, 1);
+
+      UIManager.manageChildren(
+        parentInstance._nativeTag, // containerID
+        [], // moveFromIndices
+        [], // moveToIndices
+        [], // addChildReactTags
+        [], // addAtIndices
+        [index], // removeAtIndices
+      );
+    }
+  },
+
+  resetAfterCommit() : void {
+    // Noop?
+  },
+
+  resetTextContent(instance : Instance) : void {
+    // Noop?
+  },
+
+  scheduleAnimationCallback: window.requestAnimationFrame,
+
+  scheduleDeferredCallback: window.requestIdleCallback,
+
+  shouldSetTextContent(props : Props) : boolean {
+    return false;
```

Comment:
Yeah I am not sure how much overhead the fiber and child work have in this case either. It felt kind of awkward implementing it in my PR since it was basically duplicating the createTextInstance method.